File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>

using namespace std;

int answer = 0;
void get_target_numbers(vector<int> numbers, int target, int sum, int index) {
    if(index == numbers.size()) {
        if(sum == target) answer++;
        return;
    }
    
    get_target_numbers(numbers, target, sum+numbers[index], index+1);
    get_target_numbers(numbers, target, sum-numbers[index], index+1);
}

int solution(vector<int> numbers, int target) {
    get_target_numbers(numbers, target, 0, 0);
    return answer;
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
import java.util.*;
import java.io.*;

class Solution {
    int ans = 0;
    int maxIndex;
    public int solution(int[] numbers, int target) {
        maxIndex = numbers.length - 1;
        recur(numbers, 0, 0, target);
        return ans;
    }
    public void recur(int[] numbers, int index, int sum, int target) {
        if(index > maxIndex) {

            if(sum == target) ans++;
            return;
        }
        recur(numbers, index+1, sum+numbers[index], target);
        recur(numbers, index+1, sum-numbers[index], target);
    }
}